openedx/core/djangoapps/course_roles/tests/test_helpers.py

```diff
@@ -453,3 +456,69 @@ def test_course_or_organization_list_permission_check_with_permission_in_another
         assert not course_or_organization_permission_list_check(
             self.user_1, test_permissions, self.course_1.id, self.organization_1.name
         )
+
+
+@ddt.ddt
+class GetAllUserPermissionsTestcase(SharedModuleStoreTestCase):
+    """
+    Tests of get_all_user_permissions_for_a_course function in course_roles.helpers module
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.user_1 = UserFactory(username="test_user_1")
+        self.organization_1 = OrganizationFactory(name="test_organization_1")
+        self.course_1 = CourseFactory.create(
+            display_name="test course 1", run="Testing_course_1", org=self.organization_1.name
+        )
+        self.role_1 = CourseRolesRole.objects.create(name="test_role_1")
+        self.role_2 = CourseRolesRole.objects.create(name="test_role_2")
+        self.role_3 = CourseRolesRole.objects.create(name="test_role_3")
+        self.service = CourseRolesService.objects.create(name="test_service")
+        self.role_1.services.add(self.service)
+        self.role_2.services.add(self.service)
+        self.role_3.services.add(self.service)
+        self.permission_1 = CourseRolesPermission.objects.create(name="test_permission_1")
+        self.permission_2 = CourseRolesPermission.objects.create(name="test_permission_2")
+        self.permission_3 = CourseRolesPermission.objects.create(name="test_permission_3")
+        self.role_1.permissions.add(self.permission_1)
+        self.role_2.permissions.add(self.permission_2)
+        self.role_3.permissions.add(self.permission_3)
+
+    def test_get_all_user_permissions_for_a_course(self):
```

Comment:
Please add a test that assigns roles to 2 users and ensures only the permissions for the correct user are returned by get_all_user_permissions_for_a_course